### PR TITLE
`General`: Improve the design of the course header description

### PR DESCRIPTION
**Pull request by JohannesStoehr**
#### General

- [x] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client/) and ensured that the layout is responsive.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Improve the UI of the course to make it cleaner when the course has a description.

### Description

The amount of characters displayed in the course header is now responsive depending of the window width.
The course logo is still omitted once the window gets to small.

### Steps for Testing

Check the visual appearance of the course header with short and long (>500 characters) course descriptions and with different window sizes.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage

| Class/File | Branch | Line |
|------------|-------:|-----:|
| header-course.component.ts | 100% | 100% |

### Screenshots

Old:
Bildschirmfoto 2022-06-22 um 23 42 38
Bildschirmfoto 2022-06-22 um 23 43 35
Bildschirmfoto 2022-06-22 um 23 44 02
New:
Bildschirmfoto 2022-06-22 um 23 47 58
Bildschirmfoto 2022-06-22 um 23 48 07
Bildschirmfoto 2022-06-22 um 23 49 11
Bildschirmfoto 2022-06-22 um 23 48 22

### CODE REVIEW

**Review by akesfeden — COMMENTED** [review]
Tested on TS2, works as described. I have the questions/suggestions below, I can approve according to whether they are going to be addressed in this PR or another one.
1. Can we use the space for the logo as well (even for bigger windows) if it is not set? (See the space at the right-side of the screenshot below)
2. Since the description can be long, what do you think about converting Description text input to a `textarea` where the user can also add new lines?

image

**Review by jbllhff — APPROVED** [review]
Tested on ts2, looks good, thanks!

**Review by yuling-sun — DISMISSED** [review]
Tested on TS2. Course description works as expected.

**Review by SimonPaul02 — APPROVED** [review]
Tested on Ts2, works as expected

**Review by rhubinger — DISMISSED** [review]
Tested on TS2. Works as described

**Review by alaraozdenler — DISMISSED** [review]
Tested on ts1, works as described

**Comment by ge65cer** [comment]
> 1. Can we use the space for the logo as well (even for bigger windows) if it is not set? (See the space at the right-side of the screenshot below)

That's a nice idea. Probably not too hard to implement, just remove the column if no logo is set.

**Review by ge65cer — DISMISSED** [review]
Tested on TS2, works! Code also approved.

**Review by julian-christl — CHANGES_REQUESTED** [review]
Very small remarks

**Review by babyygemperor — DISMISSED** [review]
Tested on ts2, works as described

File: src/test/javascript/spec/component/course/header-course.component.spec.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    };
+
+    beforeEach(() => {
+        TestBed.configureTestingModule({
+            providers: [HeaderCourseComponent],
+            imports: [ArtemisTestModule],
+        })
+            .compileComponents()
+            .then(() => {
+                component = TestBed.createComponent(HeaderCourseComponent).componentInstance;
```
[the remark this answers is not in the document]
Comment (JohannesStoehr) [reply src/test/javascript/spec/component/course/header-course.component.spec.ts:30]: I looked through the code. inject is used for services and controllers, createComponent for Components

**Review by Dominik-Weinzierl — APPROVED** [review]
Thanks for implementing our suggestions :+1: we talked about the text area during the testing session and i think its a good choice :+1: Code looks good to me!

**Review by ge65cer — APPROVED** [review]
Re-approve

File: src/test/javascript/spec/component/course/header-course.component.spec.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (julian-christl) [reply src/test/javascript/spec/component/course/header-course.component.spec.ts:30]: Ah, alright. Then I remembered this wrongly!

**Review by julian-christl — APPROVED** [review]
Thanks for the fix, code approved.

**Review by willergeorg — APPROVED** [review]
Tested it on TS1. Looks good on different screen sizes. 👍

**Review by babyygemperor — APPROVED** [review]
Tested on ts2 again, works as described 👍🏼

**Review by max-bergmann — APPROVED** [review]
Tested on TS1. Works as expected

**Pull request merged**